proposals/multiplatform/binary-signature.md

```diff
@@ -0,0 +1,601 @@
+declarations that are totally identical and differ only the signature name, as they cannot be distinguished by a 
+call even with parameter names:
+
+```kotlin
+// ERROR: Conflicting overloads
+fun doSomething(param: Int)
+@BinarySignatureName("doSomethingString")
+fun doSomething(param: Int): String { /* ... */ }
+```
+
+However, the same pair of declarations compiles if the first one is hidden:
+
+```kotlin
+// OK
+@Deprecated("Binary compatibility", level = DeprecationLevel.HIDDEN)
+fun doSomething(param: Int)
+@BinarySignatureName("doSomethingString")
+fun doSomething(param: Int): String { /* ... */ }
+```
+
+#### SinceKotlin/DeprecatedSinceKotlin
+
+The Kotlin standard library uses `@SinceKotlin("...")` and `@DeprecatedSinceKotlin(hiddenSince = "...")` to hide 
+declarations conditionally, based on the current API version in use (the value of the `-api-version` argument 
+when compiling the module where the usage is located). For the purposes of **call resolution** check, 
+they will taken into account, too:
+
+* Declarations with `@SinceKotlin(version)`, when the current API version is below the specified version, are excluded from the check.
+* Declarations with `@DeprecatedSinceKotlin(hiddenSince = version)`, when the current API version is the same or more recent as the version, are excluded from the check.
+
+### Override matching
+
+Override matching is performed using [**override matching signature**](#override-matching-signature). 
+It has to match between the override and the overridden method, which means the the source name, the binary signature name, 
+and all the parameter names must match or there will be a compilation error. In particular, this example is an error 
+(the source name does not match):
+
+```kotlin
+open class Base { 
+    open fun foo()
+}
+
+class Derived : Base() {
+    @SingatureName("foo")
+    override fun bar() // ERROR: 'bar' does not override anything
+}
+```
+
+And this example is an error (the signature name does not match):
+
+```kotlin
+open class Base { 
+    open fun foo()
+}
+
+class Derived : Base() {
+    @SingatureName("bar")
+    override fun foo() // ERROR: 'foo' does not override anything
+}
+```
+
+#### Overrides and properties
+
+Properties can have signature names on their getters and/or setters.
+
+* When overriding a read-only (`val`) property, the signature name for the getter must match.
+* When overriding a mutable (`var`) property, the signature name for BOTH the getter and the setter must match.
+
+#### Objective-C overrides in Kotlin/Native
+
+Objective-C has relaxed rules on overrides. In particular, it is possible to change the parameter types in Object-C on override. 
+These relaxed checks are implemented only in Kotlin/Native for Kotlin code that is produced as a result of ObjC-interop tooling and 
+their actual specifications are out of the scope of this KEEP.
```

Comment:
Objective-C is called "Objective-C" (the 1st sentence), "Object-C" (the 1st sentence) and "ObjC" (the 2nd sentence) at the same time. Is it a typo or intended?

Reply:
`Object-C` is a type, while an `ObjC` is a proper abbreviation.